actor/image_upload/README.md

```diff
@@ -0,0 +1,140 @@
+   actor, the more limited list of permissions above should be sufficient. Please consult the IAM and S3 documentation
+   for details on setting access policies.
+
+   - Create a bucket for uploading images. For this example, we use the bucket name `images.from.wasmcloud`. 
+
+     ```shell
+     aws s3 mb s3://images.from.wasmcloud
+     ```
+
+   - If you wish to use a different bucket name, edit `image_upload/Makefile` to
+   replace `images.from.wasmcloud` with your preferred bucket name.
+   The upload_actor uses an alias "images", which is mapped to the real bucket name through the link with the provider.
+
+2. **Build the capability provider and actor**
+
+   - clone `github.com/wasmcloud/capability-providers` and `github.com/wasmcloud/examples`.
+    
+   - In `capability-providers/blobstore-s3`, run `make`. 
+     If you want to run the test suite, `make test`. All tests should 
+     pass. (Ignore "ERROR nats::jetstream failed to parse API response" which is not a test error.)
+
+   - In `examples/actor/image_upload`, run `make`
+
+
+3. **Configure and Start wasmcloud**
+
+   - start nats and a local registry
+     ```shell
+     docker run -d --rm --name nats -p 127.0.0.1:4222:4222 nats:2.7.2 -js
+     docker run -d --rm --name registry -p 127.0.0.1:5000:5000 registry:2.7
+     ```
+   - set up aws credentials and environment
+     - The blobstore-s3 provider will fetch credentials from aws-standard config
+     files (~/.aws/config and ~/.aws/credentials), or the environment, or values
+     in the link definition. See blobstore-s3/README.md for more details.
+     - If you are using environment variables, you must define, at a minimum,
+     `AWS_ACCONT_KEY_ID`, `AWS_SECRET_KEY`, and `AWS_REGION`.
+       (AWS_REGION can be a multi-region endpoint if desired).
+     - also set
+     ```shell
+     export WASMCLOUD_OCI_ALLOWED_INSECURE=localhost:5000
+     export RUST_LOG=debug
+     ```
+   - [Start the wasmcloud host](https://wasmcloud.dev/overview/installation/).
+   Keep a terminal window open tailing the host
+  log.
+
+
+4. **Start capability providers and actor**
+
+   - in blobstore-s3, run `make inspect` and look for the 56-character
+   Service key starting with `V`. Copy the key and paste it into
+   `image_upload/Makefile` to replace the value for `BLOBSTORE_ID` (near line 27).
+
+   - in blobstore-s3, `make run`, and wait for it to complete.
+   - in image_upload, `make run`.
+   - If there were no startup errors, you should be ready to go.
+
+## Run
+
+The actor understands a small set of HTTP commands, which can be
+demonstrated with `curl`. To avoid leaking server details to an
+attacker, most errors return '404 not found', and you'll need to inspect
+the host logs for more detail. These commands generate output in json, so
+you might want to pipe the output to `jq` to make it more readable.
+
+### List buckets
+
+List all readable buckets.
+
+```shell
+curl localhost:8080/containers | jq
```

Comment:
my curl on linux (I realize I'm an aberration) displays timing and processing headers prior to the actual body. For me, I have to enable silent mode (`-s`) in order to get a `jq` parseable output.

Reply:
updated README